varianteval/scripts/annotation_script.py

```diff
@@ -0,0 +1,124 @@
+"""
+Script to add annotation to input callset with respect to specified
+comparison callsets
+"""
+
+import sys
+import re
+import numpy as np
+from pysam import VariantFile
+import argparse
+from collections import defaultdict
+
+
+def overlapping_records(intrvl_A, intrvl_B, overlap_threshold=0.5, bp_margin=5000):
+    '''
+    Helper method to determine whether two records' start-end
+    intervals overlap to a given breakpoint margin, and whether
+    the intervals' overlap is >=0.5*(svlen) for interval A and B
+    '''
+    bp_check = (np.abs(intrvl_A[0] - intrvl_B[0]) < bp_margin \
+                and np.abs(intrvl_A[1] - intrvl_B[1]) < bp_margin)
+    intrvl_overlap = (min(intrvl_A[1], intrvl_B[1])) - (max(intrvl_A[0], intrvl_B[0]))
+    overlap_check = (intrvl_overlap >= overlap_threshold * (intrvl_A[1] - intrvl_A[0])) and \
+                    (intrvl_overlap >= overlap_threshold * (intrvl_B[1] - intrvl_B[0]))
+
+    return bp_check and overlap_check
+
+
+def build_callset_dict(callset_vcf):
+    record_dict = defaultdict(list)
+    callset = VariantFile(callset_vcf)
+    for rec in callset.fetch():
+        record_dict[rec.chrom].append(
```

Comment:
this could be represented as an interval tree to faster overlap check (the repo has a few examples)